spec/indexers/hyrax/generic_type_indexer_spec.rb

```diff
@@ -0,0 +1,48 @@
+RSpec.describe Hyrax::GenericTypeIndexer do
+  let(:user) { create(:user) }
+  let(:admin_set) { AdminSet.new(title: ['Some admin set']) }
+  let(:work) { create_for_repository(:work) }
+  let(:collection) { Collection.new(title: ['Some collection']) }
+  let(:fs) { FileSet.new }
+
+  context "when the object is an AdminSet" do
+    let(:as_service) { described_class.new(resource: admin_set) }
+
+    subject(:as_solr_document) { as_service.to_solr }
+
+    it "indexes generic_type as Admin Set" do
+      expect(as_solr_document.fetch(:generic_type_sim)).to eq 'Admin Set'
+    end
+  end
+
+  context "when the object is a Collection" do
+    let(:col_service) { described_class.new(resource: collection) }
+
+    subject(:col_solr_document) { col_service.to_solr }
+
+    it "does not index generic_type as Collection" do
+      expect(col_solr_document.fetch(:generic_type_sim)).to eq 'Collection'
+    end
+  end
+
+  context "when the object is a Work" do
+    let(:work_service) { described_class.new(resource: work) }
+
+    subject(:work_solr_document) { work_service.to_solr }
+
+    it "does not index generic_type as Work" do
+      expect(work_solr_document.fetch(:generic_type_sim)).to eq 'Work'
+    end
+  end
+
+  context "when the object is a Fileset" do
+    let(:fs_service) { described_class.new(resource: fs) }
+
+    subject(:fs_solr_document) { fs_service.to_solr }
+
+    it "does not index generic_type" do
```

Comment:
Why not support File Set?
You could make another test for a plain old `Valkyrie::Resource` for testing an unknown model.

Reply:
In current Hyrax, FileSet doesn't have a GenericType set. I could add it ... but I'm not sure what it's return would be, 'File Set' or 'File' (as human_readable_type for file_set gets set to File). So I think I'd rather leave it as is?

Reply:
👍 makes sense.